src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -219,19 +220,44 @@ public Visitable visit(MethodDeclaration method, Void p) {
   @Override
   public Visitable visit(Parameter para, Void p) {
     if (insideTargetMethod) {
-      ResolvedType paraType = para.resolve().getType();
-      if (paraType.isReferenceType()) {
-        String paraTypeFullName =
-            paraType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
-        usedClass.add(paraTypeFullName);
-        for (ResolvedType typeParameterValue : paraType.asReferenceType().typeParametersValues()) {
-          String typeParameterValueName = typeParameterValue.describe();
-          if (typeParameterValueName.contains("<")) {
-            // removing the "<...>" part if there is any.
-            typeParameterValueName =
-                typeParameterValueName.substring(0, typeParameterValueName.indexOf("<"));
+      Type type = para.getType();
+      String paraTypeFullName = "";
+      ResolvedType paramType;
+      if (type.isUnionType()) {
+        for (var param : para.getType().asUnionType().getElements()) {
+          try {
+            paramType = param.resolve();
+            paraTypeFullName =
+                paramType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
+            usedClass.add(paraTypeFullName);
+          } catch (
+              Exception
+                  e) { // This visitor should not have an unsolvable parameter type. Throwing if
+            // exception occurs.
+            throw new Error(e.getMessage());
```

Comment:
Why catch and rethrow? If you're not going to add any information to the exception, you should just let it pass up the call stack. Adding a wrapper like this is not helpful.

Reply:
updated accordingly